akka-projection-samples/src/main/scala/akka/projection/Samples.scala

```diff
@@ -0,0 +1,168 @@
+
+  implicit val system = ActorSystem("sample")
+  implicit val ec = system.dispatcher
+
+  // we need a function that given an optional offset, returns a Source
+  def sourceProvider(offset: Option[Offset]): Source[EventEnvelope, _] = {
+    val queries = PersistenceQuery(system).readJournalFor[CassandraReadJournal](CassandraReadJournal.Identifier)
+    queries.eventsByTag("someTag", offset.getOrElse(Offset.noOffset))
+  }
+
+  val projection =
+    JdbcProjection.transactional(
+      sourceProvider = sourceProvider,
+      offsetStore = new JdbcOffsetStoreAkkaOffset(projectionId = "demo"),
+      offsetExtractor = (ev: EventEnvelope) => ev.offset) { (connec, envelope) =>
+      println(s"pretending that this is projected to some JDBC model")
+    }
+
+  projection.start()
+  projection.stop()
+}
+
+/**
+ * sample showing a rough API for consuming events from Akka Persistence JDBC and saving the
+ * projected model and the offset in a JDBC table
+ */
+object AkkaPersistenceJdbcWithTransactionalJDBCSample {
+
+  implicit val system = ActorSystem("sample")
+  implicit val ec = system.dispatcher
+
+  // we need a function that given an optional offset, returns a Source
+  def sourceProvider(offset: Option[Offset]): Source[EventEnvelope, _] = {
+    val queries = PersistenceQuery(system).readJournalFor[JdbcReadJournal](JdbcReadJournal.Identifier)
+    queries.eventsByTag("someTag", offset.getOrElse(Offset.noOffset))
+  }
+
+  val projection =
+    JdbcProjection.transactional(
+      sourceProvider = sourceProvider,
+      offsetStore = new JdbcOffsetStoreAkkaOffset(projectionId = "demo"),
+      offsetExtractor = (ev: EventEnvelope) => ev.offset) { (connec, envelope) =>
+      println(s"pretending that this is projected to some JDBC model")
+    }
+
+  projection.start()
+  projection.stop()
+}
+
+/**
+ * sample showing a rough API for consuming events from Alpakka Kafka and saving the
+ * projected model and the offset in a JDBC table
+ */
+object KafkaSourceWithTransactionalJDBCSample {
+
+  implicit val system = ActorSystem("sample")
+  implicit val ec = system.dispatcher
+
+  // we need a function that given an optional offset, returns a Source
+  def sourceProvider(offset: Option[Long]): Source[ConsumerRecord[String, String], _] = {
+    val consumerSettings: ConsumerSettings[String, String] = ???
+    val topicPartition: TopicPartition = ???
+    Consumer.plainSource(
+      consumerSettings,
+      Subscriptions.assignmentWithOffset(topicPartition -> offset.map(_ + 1).getOrElse(0L)))
+  }
+
+  val projection =
+    JdbcProjection.transactional(
+      sourceProvider = sourceProvider,
+      offsetStore = new JdbcOffsetStoreLong(projectionId = "demo"),
+      offsetExtractor = (record: ConsumerRecord[String, String]) => record.offset()) { (connec, envelope) =>
+      // whatever is
+      println(s"pretending that this is projected to some JDBC model")
+    }
+  projection.start()
```

Comment:
Here using Kafka, but could be any Source. I would consider this example as the more low level one where the user needs to know which pieces to choose. They need to choose (or build) an offset store compatible with the offset type and they need an extractor.